pkg/controller/seed-controller-manager/kubernetes/resources.go

```diff
@@ -1053,3 +1059,41 @@ func hostnameToIPList(ctx context.Context, hostname string) ([]net.IP, error) {
 
 	return ipList, nil
 }
+
+func (r *Reconciler) ensureAuditWebhook(ctx context.Context, c *kubermaticv1.Cluster, data *resources.TemplateData) error {
+	auditWebhookSettings := c.Spec.AuditLogging.WebhookBackend
+	if data.DC().Spec.EnforcedAuditWebhookSettings != nil {
+		auditWebhookSettings = data.DC().Spec.EnforcedAuditWebhookSettings
+		// if webhook backend is enabled on the DC then create the auditwebhookconfig secret in the user cluster ns.
+		creators := []reconciling.NamedSecretReconcilerFactory{r.auditWebhookSecretReconciler(ctx, data)}
+		if err := reconciling.ReconcileSecrets(ctx, creators, c.Status.NamespaceName, r.Client); err != nil {
+			return nil
+		}
+	}
+
+	return r.updateCluster(ctx, c, func(c *kubermaticv1.Cluster) {
+		c.Spec.AuditLogging.WebhookBackend = auditWebhookSettings
+	})
```

Comment:
This shouldn't happen in a reconciler (a rule of thumb: reconcilers shouldn't update a clusterSpec [we are not following this rule everywhere, but for new things we should default to it]). You already have it in `defaulting/cluster.go`, that should be sufficient,no?

Reply:
Depends on the context. The defaulting in the defaulting package is usually meant to be temporary/runtime-only and not to be persisted.

Reply:
then maybe it needs to go into the mutating webhook? But I have huge concerns with updating a spec field.